### PR TITLE
add pod monitor selectors for prometheus

### PR DESCRIPTION
**Pull request by wei-lee**
https://issues.redhat.com/browse/INTLY-4129

### CODE REVIEW

**Comment by wei-lee** [comment]
@david-martin @pb82 do we need to release a new version of the operator?

File: templates/prometheus.yaml [hunk 1]
```diff
@@ -81,6 +81,12 @@ spec:
   ruleNamespaceSelector:
     matchExpressions:
       - { key: "monitoring-key", operator: In, values: [{{ .MonitoringKey }}] }
+  podMonitorNamespaceSelector:
+    matchExpressions:
+      - { key: "monitoring-key", operator: In, values: [{{ .MonitoringKey }}] }
+  podMonitorSelector:
```
Comment (pb82) [inline templates/prometheus.yaml]: I think David was thinking about only using namespace selectors and always selecting all resources in that namespace. Can you confirm @david-martin ? Should we still have the resource selector?

Comment (david-martin) [reply templates/prometheus.yaml]: Correct, we can just add the namespace selector here.
(cc @obrienrobert who is looking at removing the serviceMonitorSelector & ruleSelector)

Comment (wei-lee) [reply templates/prometheus.yaml]: ok, I will remove it then

**Comment by wei-lee** [comment]
@pb82 @david-martin mind taking another look?

**Comment by david-martin** [comment]
@wei-lee @davidkirwan the linked jira issue https://issues.redhat.com/browse/INTLY-4129 is resolved.
Is this PR still valid?

**Comment by davidkirwan** [comment]
This looks like its still relevant @david-martin to allow us to pass in the label that the operator should watch for on PodMonitors

**Pull request merged**